### ISSUE TITLE
Best approach for removing hook from Model? 

### ISSUE THREAD

**Issue by facultymatt** [platanus/angular-restmod#256]
I have a mixin that is mixed into the `Model` api. 

I add a hook, `Scope.on('before-request', cb);`.

What is the best way to remove this callback? Does an `off` method need to be added? Would an `$off` method need to be added as well (not sure of the difference between `on` and `$on`).

**Comment by iobaixas** [comment]
`on` is a builder method, it is only available inside a definition function as is intended to be used as an alternative to the definition object's `$hooks` property. There is no way of unsubscribing a hook since is supposed to be used to modify a type behaviour.

`$on` is used to add callbacks to collections, records, etc. There is currently no `$off` method (my bad), it could be easily added by decorating the `RMCommonApi` service, like this:

```js
angular.module('bleh').config(function($provide) {
  $provide.decorator(`RMCommonApi`, function($delegate) {
    $delegate.$off = function(_hook, _cb) {
      var hooks = (this.$cb && this.$$cb[_hook]),
            idx = hooks ? hooks.indexOf(_cb) :  -1; 
      if(idx != -1) hooks.splice(idx, 1);
      return this;
    };
    return $delegate; // not sure if needed.
  });
});

```

**Comment by jpulec** [comment]
Closed by #257
